src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,8 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating())
+            storage.save(addressBook);
```

Comment:
Though the curly braces are not required, still it is encouraged to use to increase the readability of the code